integration-tests/src/test/java/org/apache/fineract/integrationtests/ClientTest.java

```diff
@@ -263,4 +282,51 @@ public void testClientAddressCreationWorksAfterClientIsCreated() {
         assertThat(addressResponse.getPostalCode()).isEqualTo(postalCode);
     }
 
+    @Test
+    public void createClientWithDatatable() {
+        // creating datatable for client entity person subentity
+        final HashMap<String, Object> columnMap = new HashMap<>();
+        final List<HashMap<String, Object>> datatableColumnsList = new ArrayList<>();
+        final String datatableName = Utils.randomNameGenerator(CLIENT_APP_TABLE_NAME + "_", 5).toLowerCase().toLowerCase();
+        columnMap.put("datatableName", datatableName);
+        columnMap.put("apptableName", CLIENT_APP_TABLE_NAME);
+        columnMap.put("entitySubType", "PERSON");
+        columnMap.put("multiRow", false);
+        String itsAString = "itsastring";
+        String dateFormat = "dateFormat";
+
+        DatatableHelper.addDatatableColumns(datatableColumnsList, itsAString, "String", true, 10, null);
+        columnMap.put("columns", datatableColumnsList);
+        String datatabelRequestJsonString = new Gson().toJson(columnMap);
+        LOG.info("map : {}", datatabelRequestJsonString);
+
+        HashMap<String, Object> datatableResponse = datatableHelper.createDatatable(datatabelRequestJsonString, "");
+        final String resourceIdentifier = (String) datatableResponse.get("resourceIdentifier");
+        DatatableHelper.verifyDatatableCreatedOnServer(requestSpec, responseSpec, resourceIdentifier);
+
+        final PostEntityDatatableChecksTemplateResponse entityPersonDatatableChecksResponse = entityDatatableChecksHelper
+                .addEntityDatatableCheck(CLIENT_APP_TABLE_NAME, datatableName, 100, null);
+        assertNotNull(entityPersonDatatableChecksResponse);
```

Comment:
Let's check more things from the response body, not just that it's not null.

Reply:
Why do I need to validate this? If the change is related to Client creation with datatable data